### ISSUE TITLE
Filter_users? Is it possible to filter a list of users from search

### ISSUE THREAD

**Issue by AndrewSB** [twintproject/twint#639]
Hey! I was wondering if there was a option to filter a set of users from search results? I'm using `twint.run.Search` and would love to be able to filter a list of users from my search results, instead of filtering them out in my client after downloading the list of tweets from twitter.

Doesn't look like the twitter search API has that feature though, so it might be impossible - thought I'd ask because I don't see an issue about it

Basically the inverse of `Members_list`

**Comment by pielco11** [comment]
May you elaborate please?

Would you like to exclude users from the search, or search for tweets sent by specific users? Or something else?

**Comment by AndrewSB** [comment]
exclude users from the search

**Comment by pielco11** [comment]
`twint -s "keyword -from:excluded_username"`

**Comment by AndrewSB** [comment]
sorry, i'm using the python API. how does that translate? Do I just append that into `twint.Config().Search`, or is there another config key I should put that in?

**Comment by pielco11** [comment]
Do you mean Twint as module? About the API, better check the official documentation

**Comment by TurquoiseA** [comment]
> Do you mean Twint as module? About the API, better check the official documentation

Hello pielco11, I think he meant using twint as a module in python. In that case, suppose I have a list of users, and I don't want their tweets to be included in the result since they are just bots, how can we achieve this. 

Many thanks.

**Comment by himanshudabas** [comment]
@TurquoiseA 
Do something like this:
```
c = twint.Config()
c.Search = "india -from:realDonaldTrump -from:elonmusk"
twint.run.Search(c)
```

Here **india** is the keyword which will be searched for, and this will exclude tweets from **realDonaldTrump** & **elonmusk**.

_**Note**: there is a limit to how many users you can add to this exclusion list.
I don't know what that limit is, I haven't tested it myself, but you can test that on your own._

If you have a `list` of users you want to exclude, let's say **exclude_list**, then you can make the exclusion list  by doing something like this:
```
search_filter = None
for user in exclude_list:
    search_filter += f"-from:{user} "
c = twint.Config()
c.Search = "india " + search_filter
twint.run.Search(c)
```

Here search_filter is of `list` type and contains the usernames of the profiles you want to exclude.

**Comment by TurquoiseA** [comment]
@himanshudabas 
Thank you for your quick reply. It works perfectly for me. Just one more question: Is it possible to use a combination of strings as one keyword? 

For example, the result must have all 4 words `Resolution for Indian Farmers` or the keyword `Indian` included. I tried the following code but it doesn't seem to work:

```
c.Search = "(Resolution for Indian Farmers) OR Indian"
```

I guess the query rules are different from twitter api

**Comment by himanshudabas** [comment]
@TurquoiseA 
actually the rules are exactly the same as **Twitter Advanced Search**.
It's works for me as intended.

To test it, you can first run the same exact query in **Twitter Advanced Search** and compare it with _twint_ results.

Check [this](https://twitter.com/search?q=(Resolution%20for%20Indian%20Farmers)%20OR%20Indian&src=typed_query) out.
Also compare the results of the above search results with the following script:

```
import twint

c = twint.Config()

c.Search = "(Resolution for Indian Farmers) OR Indian"
c.Store_csv = True
c.Output = "advanced_query.csv"
c.Limit = 200

twint.run.Search(c)
```
